### ISSUE TITLE
lxc exec / shell triggers POLLNVAL event 

### ISSUE THREAD

**Issue by wmchris** [lxc/lxd#9036]
problem just appeared - lxc exec container bash simply does nothing anymore. all other (common) commands work flawless

Base system: proxmox 7 (debian 11)
Kernel: Linux pve 5.11.22-2-pve #1 SMP PVE 5.11.22-4 (Tue, 20 Jul 2021 21:40:02 +0200) x86_64 GNU/Linux
LXD 4.16 installed via snap
lxd error:
`Detected poll(POLLNVAL) event`

lxc info:
```
config:
  core.https_address: 10.0.10.1
  core.trust_password: true
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- candid_authentication
- backup_compression
- candid_config
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- candid_config_key
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- snapshot_schedule_aliases
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
- snapshot_expiry_creation
- network_leases_location
- resources_cpu_socket
- resources_gpu
- resources_numa
- kernel_features
- id_map_current
- event_location
- storage_api_remote_volume_snapshots
- network_nat_address
- container_nic_routes
- rbac
- cluster_internal_copy
- seccomp_notify
- lxc_features
- container_nic_ipvlan
- network_vlan_sriov
- storage_cephfs
- container_nic_ipfilter
- resources_v2
- container_exec_user_group_cwd
- container_syscall_intercept
- container_disk_shift
- storage_shifted
- resources_infiniband
- daemon_storage
- instances
- image_types
- resources_disk_sata
- clustering_roles
- images_expiry
- resources_network_firmware
- backup_compression_algorithm
- ceph_data_pool_name
- container_syscall_intercept_mount
- compression_squashfs
- container_raw_mount
- container_nic_routed
- container_syscall_intercept_mount_fuse
- container_disk_ceph
- virtual-machines
- image_profiles
- clustering_architecture
- resources_disk_id
- storage_lvm_stripes
- vm_boot_priority
- unix_hotplug_devices
- api_filtering
- instance_nic_network
- clustering_sizing
- firewall_driver
- projects_limits
- container_syscall_intercept_hugetlbfs
- limits_hugepages
- container_nic_routed_gateway
- projects_restrictions
- custom_volume_snapshot_expiry
- volume_snapshot_scheduling
- trust_ca_certificates
- snapshot_disk_usage
- clustering_edit_roles
- container_nic_routed_host_address
- container_nic_ipvlan_gateway
- resources_usb_pci
- resources_cpu_threads_numa
- resources_cpu_core_die
- api_os
- container_nic_routed_host_table
- container_nic_ipvlan_host_table
- container_nic_ipvlan_mode
- resources_system
- images_push_relay
- network_dns_search
- container_nic_routed_limits
- instance_nic_bridged_vlan
- network_state_bond_bridge
- usedby_consistency
- custom_block_volumes
- clustering_failure_domains
- resources_gpu_mdev
- console_vga_type
- projects_limits_disk
- network_type_macvlan
- network_type_sriov
- container_syscall_intercept_bpf_devices
- network_type_ovn
- projects_networks
- projects_networks_restricted_uplinks
- custom_volume_backup
- backup_override_name
- storage_rsync_compression
- network_type_physical
- network_ovn_external_subnets
- network_ovn_nat
- network_ovn_external_routes_remove
- tpm_device_type
- storage_zfs_clone_copy_rebase
- gpu_mdev
- resources_pci_iommu
- resources_network_usb
- resources_disk_address
- network_physical_ovn_ingress_mode
- network_ovn_dhcp
- network_physical_routes_anycast
- projects_limits_instances
- network_state_vlan
- instance_nic_bridged_port_isolation
- instance_bulk_state_change
- network_gvrp
- instance_pool_move
- gpu_sriov
- pci_device_type
- storage_volume_state
- network_acl
- migration_stateful
- disk_state_quota
- storage_ceph_features
- projects_compression
- projects_images_remote_cache_expiry
- certificate_project
- network_ovn_acl
- projects_images_auto_update
- projects_restricted_cluster_target
- images_default_architecture
- network_ovn_acl_defaults
- gpu_mig
- project_usage
- network_bridge_acl
- warnings
- projects_restricted_backups_and_snapshots
- clustering_join_token
- clustering_description
- server_trusted_proxy
- clustering_update_cert
- storage_api_project
- server_instance_driver_operational
- server_supported_storage_drivers
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses:
  - 10.0.10.1:8443
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIIB+jCCAYCgAwIBAgIQM8cf825+q1xHTdebHJpjCjAKBggqhkjOPQQDAzAxMRww
    GgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMREwDwYDVQQDDAhyb290QHB2ZTAe
    Fw0yMDAyMjkxOTM0MjFaFw0zMDAyMjYxOTM0MjFaMDExHDAaBgNVBAoTE2xpbnV4
    Y29udGFpbmVycy5vcmcxETAPBgNVBAMMCHJvb3RAcHZlMHYwEAYHKoZIzj0CAQYF
    K4EEACIDYgAEm0NGWcXMHYVn4DLUZ8kxCSPuiXe/oafKA6nH92hiHnlt4rA5tz2w
    ZXaUjfdK7taQIULzbksvdgWiipuTS+JrmoUX6ZaGydX1sdB5FyZJs1cUzThOJmEZ
    54Z6/xUa3zCeo10wWzAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUH
    AwEwDAYDVR0TAQH/BAIwADAmBgNVHREEHzAdggNwdmWHBH8AAAGHEAAAAAAAAAAA
    AAAAAAAAAAEwCgYIKoZIzj0EAwMDaAAwZQIwfoeI9EUgWiGDm8i6QzVchEZdKjd1
    u7G9yAOd6pEyz0RMDDmSSKWNh2UgPn1O9tRpAjEA8dcqL8Vngvbyrmk7udED7FYs
    UqMUX+9vKhv+IvhQIdC339FqaJtek9um+RR7Xr1c
    -----END CERTIFICATE-----
  certificate_fingerprint: 05736f3d92355dc8e6ece598917a8abafb194bc1882784e8e0cfc129f5e83326
  driver: lxc | qemu
  driver_version: 4.0.10 | 5.2.0
  firewall: nftables
  kernel: Linux
  kernel_architecture: x86_64
  kernel_features:
    netnsid_getifaddrs: "true"
    seccomp_listener: "true"
    seccomp_listener_continue: "true"
    shiftfs: "false"
    uevent_injection: "true"
    unpriv_fscaps: "true"
  kernel_version: 5.11.22-2-pve
  lxc_features:
    cgroup2: "true"
    devpts_fd: "true"
    idmapped_mounts_v2: "true"
    mount_injection_file: "true"
    network_gateway_device_route: "true"
    network_ipvlan: "true"
    network_l2proxy: "true"
    network_phys_macvlan_mtu: "true"
    network_veth_router: "true"
    pidfd: "true"
    seccomp_allow_deny_syntax: "true"
    seccomp_notify: "true"
    seccomp_proxy_send_notify_fd: "true"
  os_name: Debian GNU/Linux
  os_version: "11"
  project: default
  server: lxd
  server_clustered: false
  server_name: pve
  server_pid: 6508
  server_version: "4.16"
  storage: zfs
  storage_version: 2.0.5-pve1
  storage_supported_drivers:
  - name: btrfs
    version: 5.4.1
    remote: false
  - name: cephfs
    version: 15.2.13
    remote: true
  - name: dir
    version: "1"
    remote: false
  - name: lvm
    version: 2.03.07(2) (2019-11-30) / 1.02.167 (2019-11-30) / 4.43.0
    remote: false
  - name: zfs
    version: 2.0.5-pve1
    remote: false
  - name: ceph
    version: 15.2.13
    remote: true
```

lxc exec container bash --debug=true
```
krim@pve:~$ lxc exec solarview bash --debug=true
WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement
DBUG[07-23|00:14:27] Connecting to a local LXD over a Unix socket 
DBUG[07-23|00:14:27] Sending request to LXD                   method=GET url=http://unix.socket/1.0 etag=
DBUG[07-23|00:14:27] Got response struct from LXD 
DBUG[07-23|00:14:27] 
	{
		"config": {
			"core.https_address": "10.0.10.1",
			"core.trust_password": true
		},
		"api_extensions": [
			"storage_zfs_remove_snapshots",
			"container_host_shutdown_timeout",
			"container_stop_priority",
			"container_syscall_filtering",
			"auth_pki",
			"container_last_used_at",
			"etag",
			"patch",
			"usb_devices",
			"https_allowed_credentials",
			"image_compression_algorithm",
			"directory_manipulation",
			"container_cpu_time",
			"storage_zfs_use_refquota",
			"storage_lvm_mount_options",
			"network",
			"profile_usedby",
			"container_push",
			"container_exec_recording",
			"certificate_update",
			"container_exec_signal_handling",
			"gpu_devices",
			"container_image_properties",
			"migration_progress",
			"id_map",
			"network_firewall_filtering",
			"network_routes",
			"storage",
			"file_delete",
			"file_append",
			"network_dhcp_expiry",
			"storage_lvm_vg_rename",
			"storage_lvm_thinpool_rename",
			"network_vlan",
			"image_create_aliases",
			"container_stateless_copy",
			"container_only_migration",
			"storage_zfs_clone_copy",
			"unix_device_rename",
			"storage_lvm_use_thinpool",
			"storage_rsync_bwlimit",
			"network_vxlan_interface",
			"storage_btrfs_mount_options",
			"entity_description",
			"image_force_refresh",
			"storage_lvm_lv_resizing",
			"id_map_base",
			"file_symlinks",
			"container_push_target",
			"network_vlan_physical",
			"storage_images_delete",
			"container_edit_metadata",
			"container_snapshot_stateful_migration",
			"storage_driver_ceph",
			"storage_ceph_user_name",
			"resource_limits",
			"storage_volatile_initial_source",
			"storage_ceph_force_osd_reuse",
			"storage_block_filesystem_btrfs",
			"resources",
			"kernel_limits",
			"storage_api_volume_rename",
			"macaroon_authentication",
			"network_sriov",
			"console",
			"restrict_devlxd",
			"migration_pre_copy",
			"infiniband",
			"maas_network",
			"devlxd_events",
			"proxy",
			"network_dhcp_gateway",
			"file_get_symlink",
			"network_leases",
			"unix_device_hotplug",
			"storage_api_local_volume_handling",
			"operation_description",
			"clustering",
			"event_lifecycle",
			"storage_api_remote_volume_handling",
			"nvidia_runtime",
			"container_mount_propagation",
			"container_backup",
			"devlxd_images",
			"container_local_cross_pool_handling",
			"proxy_unix",
			"proxy_udp",
			"clustering_join",
			"proxy_tcp_udp_multi_port_handling",
			"network_state",
			"proxy_unix_dac_properties",
			"container_protection_delete",
			"unix_priv_drop",
			"pprof_http",
			"proxy_haproxy_protocol",
			"network_hwaddr",
			"proxy_nat",
			"network_nat_order",
			"container_full",
			"candid_authentication",
			"backup_compression",
			"candid_config",
			"nvidia_runtime_config",
			"storage_api_volume_snapshots",
			"storage_unmapped",
			"projects",
			"candid_config_key",
			"network_vxlan_ttl",
			"container_incremental_copy",
			"usb_optional_vendorid",
			"snapshot_scheduling",
			"snapshot_schedule_aliases",
			"container_copy_project",
			"clustering_server_address",
			"clustering_image_replication",
			"container_protection_shift",
			"snapshot_expiry",
			"container_backup_override_pool",
			"snapshot_expiry_creation",
			"network_leases_location",
			"resources_cpu_socket",
			"resources_gpu",
			"resources_numa",
			"kernel_features",
			"id_map_current",
			"event_location",
			"storage_api_remote_volume_snapshots",
			"network_nat_address",
			"container_nic_routes",
			"rbac",
			"cluster_internal_copy",
			"seccomp_notify",
			"lxc_features",
			"container_nic_ipvlan",
			"network_vlan_sriov",
			"storage_cephfs",
			"container_nic_ipfilter",
			"resources_v2",
			"container_exec_user_group_cwd",
			"container_syscall_intercept",
			"container_disk_shift",
			"storage_shifted",
			"resources_infiniband",
			"daemon_storage",
			"instances",
			"image_types",
			"resources_disk_sata",
			"clustering_roles",
			"images_expiry",
			"resources_network_firmware",
			"backup_compression_algorithm",
			"ceph_data_pool_name",
			"container_syscall_intercept_mount",
			"compression_squashfs",
			"container_raw_mount",
			"container_nic_routed",
			"container_syscall_intercept_mount_fuse",
			"container_disk_ceph",
			"virtual-machines",
			"image_profiles",
			"clustering_architecture",
			"resources_disk_id",
			"storage_lvm_stripes",
			"vm_boot_priority",
			"unix_hotplug_devices",
			"api_filtering",
			"instance_nic_network",
			"clustering_sizing",
			"firewall_driver",
			"projects_limits",
			"container_syscall_intercept_hugetlbfs",
			"limits_hugepages",
			"container_nic_routed_gateway",
			"projects_restrictions",
			"custom_volume_snapshot_expiry",
			"volume_snapshot_scheduling",
			"trust_ca_certificates",
			"snapshot_disk_usage",
			"clustering_edit_roles",
			"container_nic_routed_host_address",
			"container_nic_ipvlan_gateway",
			"resources_usb_pci",
			"resources_cpu_threads_numa",
			"resources_cpu_core_die",
			"api_os",
			"container_nic_routed_host_table",
			"container_nic_ipvlan_host_table",
			"container_nic_ipvlan_mode",
			"resources_system",
			"images_push_relay",
			"network_dns_search",
			"container_nic_routed_limits",
			"instance_nic_bridged_vlan",
			"network_state_bond_bridge",
			"usedby_consistency",
			"custom_block_volumes",
			"clustering_failure_domains",
			"resources_gpu_mdev",
			"console_vga_type",
			"projects_limits_disk",
			"network_type_macvlan",
			"network_type_sriov",
			"container_syscall_intercept_bpf_devices",
			"network_type_ovn",
			"projects_networks",
			"projects_networks_restricted_uplinks",
			"custom_volume_backup",
			"backup_override_name",
			"storage_rsync_compression",
			"network_type_physical",
			"network_ovn_external_subnets",
			"network_ovn_nat",
			"network_ovn_external_routes_remove",
			"tpm_device_type",
			"storage_zfs_clone_copy_rebase",
			"gpu_mdev",
			"resources_pci_iommu",
			"resources_network_usb",
			"resources_disk_address",
			"network_physical_ovn_ingress_mode",
			"network_ovn_dhcp",
			"network_physical_routes_anycast",
			"projects_limits_instances",
			"network_state_vlan",
			"instance_nic_bridged_port_isolation",
			"instance_bulk_state_change",
			"network_gvrp",
			"instance_pool_move",
			"gpu_sriov",
			"pci_device_type",
			"storage_volume_state",
			"network_acl",
			"migration_stateful",
			"disk_state_quota",
			"storage_ceph_features",
			"projects_compression",
			"projects_images_remote_cache_expiry",
			"certificate_project",
			"network_ovn_acl",
			"projects_images_auto_update",
			"projects_restricted_cluster_target",
			"images_default_architecture",
			"network_ovn_acl_defaults",
			"gpu_mig",
			"project_usage",
			"network_bridge_acl",
			"warnings",
			"projects_restricted_backups_and_snapshots",
			"clustering_join_token",
			"clustering_description",
			"server_trusted_proxy",
			"clustering_update_cert",
			"storage_api_project",
			"server_instance_driver_operational",
			"server_supported_storage_drivers"
		],
		"api_status": "stable",
		"api_version": "1.0",
		"auth": "trusted",
		"public": false,
		"auth_methods": [
			"tls"
		],
		"environment": {
			"addresses": [
				"10.0.10.1:8443"
			],
			"architectures": [
				"x86_64",
				"i686"
			],
			"certificate": "-----BEGIN CERTIFICATE-----\nMIIB+jCCAYCgAwIBAgIQM8cf825+q1xHTdebHJpjCjAKBggqhkjOPQQDAzAxMRww\nGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMREwDwYDVQQDDAhyb290QHB2ZTAe\nFw0yMDAyMjkxOTM0MjFaFw0zMDAyMjYxOTM0MjFaMDExHDAaBgNVBAoTE2xpbnV4\nY29udGFpbmVycy5vcmcxETAPBgNVBAMMCHJvb3RAcHZlMHYwEAYHKoZIzj0CAQYF\nK4EEACIDYgAEm0NGWcXMHYVn4DLUZ8kxCSPuiXe/oafKA6nH92hiHnlt4rA5tz2w\nZXaUjfdK7taQIULzbksvdgWiipuTS+JrmoUX6ZaGydX1sdB5FyZJs1cUzThOJmEZ\n54Z6/xUa3zCeo10wWzAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUH\nAwEwDAYDVR0TAQH/BAIwADAmBgNVHREEHzAdggNwdmWHBH8AAAGHEAAAAAAAAAAA\nAAAAAAAAAAEwCgYIKoZIzj0EAwMDaAAwZQIwfoeI9EUgWiGDm8i6QzVchEZdKjd1\nu7G9yAOd6pEyz0RMDDmSSKWNh2UgPn1O9tRpAjEA8dcqL8Vngvbyrmk7udED7FYs\nUqMUX+9vKhv+IvhQIdC339FqaJtek9um+RR7Xr1c\n-----END CERTIFICATE-----\n",
			"certificate_fingerprint": "05736f3d92355dc8e6ece598917a8abafb194bc1882784e8e0cfc129f5e83326",
			"driver": "lxc | qemu",
			"driver_version": "4.0.10 | 5.2.0",
			"firewall": "nftables",
			"kernel": "Linux",
			"kernel_architecture": "x86_64",
			"kernel_features": {
				"netnsid_getifaddrs": "true",
				"seccomp_listener": "true",
				"seccomp_listener_continue": "true",
				"shiftfs": "false",
				"uevent_injection": "true",
				"unpriv_fscaps": "true"
			},
			"kernel_version": "5.11.22-2-pve",
			"lxc_features": {
				"cgroup2": "true",
				"devpts_fd": "true",
				"idmapped_mounts_v2": "true",
				"mount_injection_file": "true",
				"network_gateway_device_route": "true",
				"network_ipvlan": "true",
				"network_l2proxy": "true",
				"network_phys_macvlan_mtu": "true",
				"network_veth_router": "true",
				"pidfd": "true",
				"seccomp_allow_deny_syntax": "true",
				"seccomp_notify": "true",
				"seccomp_proxy_send_notify_fd": "true"
			},
			"os_name": "Debian GNU/Linux",
			"os_version": "11",
			"project": "default",
			"server": "lxd",
			"server_clustered": false,
			"server_name": "pve",
			"server_pid": 6508,
			"server_version": "4.16",
			"storage": "zfs",
			"storage_version": "2.0.5-pve1",
			"storage_supported_drivers": [
				{
					"Name": "btrfs",
					"Version": "5.4.1",
					"Remote": false
				},
				{
					"Name": "cephfs",
					"Version": "15.2.13",
					"Remote": true
				},
				{
					"Name": "dir",
					"Version": "1",
					"Remote": false
				},
				{
					"Name": "lvm",
					"Version": "2.03.07(2) (2019-11-30) / 1.02.167 (2019-11-30) / 4.43.0",
					"Remote": false
				},
				{
					"Name": "zfs",
					"Version": "2.0.5-pve1",
					"Remote": false
				},
				{
					"Name": "ceph",
					"Version": "15.2.13",
					"Remote": true
				}
			]
		}
	} 
DBUG[07-23|00:14:27] Connected to the websocket: ws://unix.socket/1.0/events 
                                                                             DBUG[07-23|00:14:27] Sending request to LXD                   method=POST url=http://unix.socket/1.0/instances/solarview/exec etag=
                                                DBUG[07-23|00:14:27] 
                                                                     	{
                                                                         	"command": [
           			"bash"
                                      		],
                                                  		"wait-for-websocket": true,
           		"interactive": true,
                                            		"environment": {
                                                                        	"TERM": "xterm-256color"
                       		},
                                  		"width": 80,
                                                            		"height": 24,
     		"record-output": false,
                                       		"user": 0,
                                                          		"group": 0,
   		"cwd": ""
                         	} 
                                  DBUG[07-23|00:14:27] Got operation from LXD 
                                                                              DBUG[07-23|00:14:27] 
                   	{
                         		"id": "01f294fa-3997-418a-8937-1297af923505",
     		"class": "websocket",
                                     		"description": "Executing command",
   		"created_at": "2021-07-23T00:14:27.973706175+02:00",
                                                                    		"updated_at": "2021-07-23T00:14:27.973706175+02:00",
                                                   		"status": "Running",
    		"status_code": 103,
                                   		"resources": {
                                                              			"containers": [
              				"/1.0/containers/solarview"
                                                                   		],
 			"instances": [
                                      				"/1.0/instances/solarview"
          			]
                                 		},
                                                  		"metadata": {
                                                                             	"command": [
           				"bash"
                                              			],
                                                                  		"environment": {
               				"HOME": "/root",
                                                        			"LANG": "C.UTF-8",
                 				"PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                                       				"TERM": "xterm-256color",
         				"USER": "root"
                                                      			},
                                                                          	"fds": {
       				"0": "cafc62f608885c52880219a820dc40c7c0c9d478725cbc6ab8fa32274e87145d",
                        				"control": "f78985479f7b08893763acfc3b7e0186e98467900b4de951c78739206b2b5ae7"
                                                     			},
                                                                          	"interactive": true
                  		},
                                  		"may_cancel": false,
                                                                    		"err": "",
         		"location": "none"
                                          	} 
                                                  DBUG[07-23|00:14:27] Connected to the websocket: ws://unix.socket/1.0/operations/01f294fa-3997-418a-8937-1297af923505/websocket?secret=f78985479f7b08893763acfc3b7e0186e98467900b4de951c78739206b2b5ae7 
          DBUG[07-23|00:14:27] Connected to the websocket: ws://unix.socket/1.0/operations/01f294fa-3997-418a-8937-1297af923505/websocket?secret=cafc62f608885c52880219a820dc40c7c0c9d478725cbc6ab8fa32274e87145d 
                                                  DBUG[07-23|00:14:27] Sending request to LXD                   method=GET url=http://unix.socket/1.0/operations/01f294fa-3997-418a-8937-1297af923505 etag=
                                           DBUG[07-23|00:14:27] Got response struct from LXD 
             DBUG[07-23|00:14:27] 
                                  	{
                                         		"id": "01f294fa-3997-418a-8937-1297af923505",
                     		"class": "websocket",
                                                     		"description": "Executing command",
                   		"created_at": "2021-07-23T00:14:27.973706175+02:00",
    		"updated_at": "2021-07-23T00:14:27.973706175+02:00",
                                                                    		"status": "Running",
                   		"status_code": 103,
                                                   		"resources": {
                                                                              	"containers": [
              				"/1.0/containers/solarview"
                                                                   		],
 			"instances": [
                                      				"/1.0/instances/solarview"
          			]
                                 		},
                                                  		"metadata": {
                                                                             	"command": [
           				"bash"
                                              			],
                                                                  		"environment": {
               				"HOME": "/root",
                                                        			"LANG": "C.UTF-8",
                 				"PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                                       				"TERM": "xterm-256color",
         				"USER": "root"
                                                      			},
                                                                          	"fds": {
       				"0": "cafc62f608885c52880219a820dc40c7c0c9d478725cbc6ab8fa32274e87145d",
                        				"control": "f78985479f7b08893763acfc3b7e0186e98467900b4de951c78739206b2b5ae7"
                                                     			},
                                                                          	"interactive": true
                  		},
                                  		"may_cancel": false,
                                                                    		"err": "",
         		"location": "none"
                                          	} 
                                                  DBUG[07-23|00:14:28] Got message barrier 
```

**Comment by stgraber** [comment]
Is it stuck waiting at that point or just returning immediately?

**Comment by wmchris** [comment]
the command finishes immediately

**Comment by wmchris** [comment]
after setting up a completly fresh system - issue appears at the moment the kernel 5.11.22-2-pve is installed, 5.11.22-1-pve works flawless. i'm not sure if this is an issue with LXC/LXD, Proxmox Kernel Modifications or an error in the kernel itself.

**Comment by stgraber** [comment]
Those aren't Ubuntu or Debian kernel versions, likely Proxmox are doing their own thing and may have picked up the same bad commit Debian did and missed the follow-up fixes.

Given reverting to a previous bugfix release fixes it, it's a kernel regression and you should reach out to the Proxmox folks to have this sorted out. We have daily tests on the current mainline kernel so we can catch those kind of issues pretty quickly, but when something breaks, it can happen that distros miss it and include the bad change without the follow-up fixes...

**Comment by stgraber** [comment]
@brauner do you have link to LKML for this handy?

**Comment by wmchris** [comment]
well the kernel itself is based on ubuntu hirsute 5.11.0-24.25

**Comment by stgraber** [comment]
I know that the Ubuntu kernel team temporarily broke our kernels too, though it got caught in the -proposed kernel and then fixed prior to it hitting users, I believe this was the 5.11 kernel.

It's worth noting that if your kernel is in fact based on 5.11.0-24.25, then it is missing the CVE fix as that's only in 5.11.0-25.27

**Comment by wmchris** [comment]
i tested it with the version 5.11.22-5 (in the test repository) and its working there again.

**Comment by ThomasLamprecht** [comment]
> It's worth noting that if your kernel is in fact based on 5.11.0-24.25, then it is missing the CVE fix as that's only in 5.11.0-25.27

Just for the record, that was patched separately to allow fast rollout:
https://git.proxmox.com/?p=pve-kernel.git;a=commitdiff;h=97b871128f3d409e41b9ee667b8751bbdaffec6c

We then updated to base off Ubuntu-5.11.0-26.28 afterwards doing a slow rollout with that one.
